pkgs/development/libraries/rocm-opencl-icd/default.nix

```diff
@@ -11,6 +11,8 @@ stdenv.mkDerivation rec {
     echo "${rocm-opencl-runtime}/lib/libamdocl64.so" > $out/etc/OpenCL/vendors/amdocl64.icd
   '';
 
+  passthru.tests = { inherit (nixosTests) rocm-opencl; };
```

Comment:
I could imagine a new attribute called `impureTests`

This attribute would return scripts that can be run outside of the nix sandbox:

```
$(nix-build -A rocm-opencl-icd.impureTests.rocm-opencl)
```